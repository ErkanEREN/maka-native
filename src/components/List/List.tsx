import React, { ReactNode } from "react";
import { FlatList } from "react-native";
import ListItem from "./ListItem/ListItem";

import { TouchableHighlight } from "react-native";
import styled from "styled-components";

interface Props {
  data: Array<object>;
  style?: {};
  RenderedElement?: any;
}

export default function List({ RenderedElement, data, style }: Props) {
  return (
    <FlatList
      data={data}
      style={style}
      renderItem={({ item }) =>
        RenderedElement ? (
          <RenderedElement item={item} />
        ) : (
          <ListItem item={item} />
        )
      }
    />
  );
}
