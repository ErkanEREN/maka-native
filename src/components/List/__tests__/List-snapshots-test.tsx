import React from "react";
import renderer from "react-test-renderer";

import List from "../List";

describe("List", () => {
  describe("Rendering", () => {
    it("should match to snapshot", () => {
      const data = [
        {
          key: 1,
          value: "1"
        },
        {
          key: 2,
          value: "2"
        }
      ];
      const tree = renderer.create(<List data={data} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
