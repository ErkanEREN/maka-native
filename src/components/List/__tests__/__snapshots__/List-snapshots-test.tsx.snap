// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Rendering should match to snapshot 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "key": 1,
        "value": "1",
      },
      Object {
        "key": 2,
        "value": "2",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        {"key":1,"value":"1"}
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        {"key":2,"value":"2"}
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
